js/dailyshine.js

```diff
@@ -462,41 +463,58 @@
   }
 
   /**
+   * Helper function to format date to MM-dd-yyyy.
+   *
+   * @param Date object
+   * @return formatted date string
+   */
+  function formatDateMMddyyy(date) {
+    var year = date.getFullYear();
+    var month = date.getMonth() + 1 >= 10 ? date.getMonth() + 1 : '0' + (date.getMonth() + 1);
+    var day = date.getDate() >= 10 ? date.getDate() : '0' + date.getDate();
+    return month + '-' + day + '-' + year;
+  }
+  // Note: for some reason `new Date('MM-dd-YYYY')` works, but
+  //   `new Date('YYYY-MM-dd')` sets date a day behind.
+
+  /**
+   * Helper function to get date of Shine
+   * @return formatted date string (mm-dd-yyyy)
+   */
+  function getShineDate() {
+    var dateObj;
+    var dateSpecified = getParameter('date');
+    if (dateSpecified) {
+      dateObj = createDateFromQuery(dateSpecified);
+      return formatDateMMddyyy(dateObj);
+    } else {
+      return formatDateMMddyyy(new Date());
+    }
+  }
+
+  /**
    * Logic to run once the end of the messaging flow has been reached.
    *
    * @param displayDelay
    */
   function onMessagesFinished(displayDelay) {
     var data;
-    var date;
     var dateQuery;
     var endMessage;
     var html;
     var template;
-    var month, day, year;
 
     // Display the end CTA section
     template = $('#template-daily-end').html();
 
     // Set the date query string for the share link
-    if (getParameter('date')) {
-      date = createDateFromQuery(getParameter('date'));
-    }
-    else {
-      date = new Date();
-    }
-
-    year = date.getFullYear();
-    month = date.getMonth() + 1 >= 10 ? date.getMonth() + 1 : '0' + (date.getMonth() + 1);
-    day = date.getDate() >= 10 ? date.getDate() : '0' + date.getDate();
-    dateQuery = month + '-' + day + '-' + year;
-    // Note: for some reason `new Date('MM-dd-YYYY')` works, but
-    //   `new Date('YYYY-MM-dd')` sets date a day behind.
+    dateQuery = getShineDate();
 
     data = {
       // Used for the share link
       shareLink: 'http://' + window.location.hostname + '?date=' + dateQuery,
       showSignUp: getParameter('r') ? false : true,
+      shineDate: dateQuery
```

Comment:
@jonuy: not sure if you intended the date for sign-up click to be treated any differently from article-click -- I was also initially thinking it would be the date the user hit 'Sign up' (in which case I think it would be the current day?), but here I've also changed it to the date of that particular Shine content that motivated her to sign up.  Let me know if this is not the case :)